staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics_test.go

```diff
@@ -85,15 +85,46 @@ func TestObserveAdmissionController(t *testing.T) {
 func TestObserveWebhook(t *testing.T) {
 	defer Metrics.reset()
 	defer legacyregistry.Reset()
-	Metrics.ObserveWebhook(context.TODO(), 2*time.Second, false, attr, stepAdmit, "x")
-	wantLabels := map[string]string{
+	Metrics.ObserveWebhook(context.TODO(), 2*time.Second, false, attr, stepAdmit, 200, "x")
+	Metrics.ObserveWebhook(context.TODO(), 2*time.Second, true, attr, stepAdmit, 500, "x")
+	Metrics.ObserveWebhook(context.TODO(), 2*time.Second, false, attr, stepValidate, 654, "x")
+	Metrics.ObserveWebhook(context.TODO(), 2*time.Second, false, attr, stepValidate, 0, "x")
+	wantLabels200 := map[string]string{
+		"name":      "x",
+		"operation": string(admission.Create),
+		"type":      "admit",
+		"rejected":  "false",
+		"namespace": "ns",
+		"code":      "200",
+	}
+	wantLabels500 := map[string]string{
 		"name":      "x",
 		"operation": string(admission.Create),
 		"type":      "admit",
+		"rejected":  "true",
+		"namespace": "ns",
+		"code":      "500",
+	}
+	wantLabels654 := map[string]string{
+		"name":      "x",
+		"operation": string(admission.Create),
+		"type":      "validate",
+		"rejected":  "false",
+		"namespace": "ns",
+		"code":      "654",
+	}
+	wantLabels000 := map[string]string{
+		"name":      "x",
+		"operation": string(admission.Create),
+		"type":      "validate",
 		"rejected":  "false",
 		"namespace": "ns",
+		"code":      "0",
```

Comment:
0->200

Reply:
This case corresponds to the WebhookRejectionAPIServerInternalError case, which does return code 0 (i.e. no http code was set).